lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,1608 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: cloudfront_distribution
+short_description: create, update, delete, duplicate and validate aws cloudfront distributions
+description:
+    - Allows for easy creation, updating, deletion, duplication and validation of CloudFront distributions.
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+version_added: "2.4"
+author: Willem van Ketwich (@wilvk)
+options:
+    state:
+      description:
+        - The desired state of the distribution or streaming distribution.
+          present - creates a new distribution
+          absent - deletes an existing distribution
+          updated - updates an existing distribution
+          duplicated - duplicates an existing distribution
```

Comment:
Yeah - I get the idea of "take a config and use it" but I think that's a touchy enough project that I'd rather have users transfer stuff either through facts or manually, since I can see a lot of weird edge cases being a problem with a "duplicate" command.